presto-main/src/main/java/com/facebook/presto/metadata/SqlScalarFunction.java

```diff
@@ -156,18 +89,16 @@ public SimpleSqlScalarFunction(
                 Optional<MethodHandle> instanceFactory,
                 boolean deterministic,
                 boolean nullable,
-                List<Boolean> nullableArguments,
-                Set<String> literalParameters)
+                List<Boolean> nullableArguments)
         {
-            super(signature.getName(),
+            super(new Signature(
```

Comment:
why do you create new signature instance instead of passing the one from argument?